cranelift/interpreter/src/environment.rs

```diff
@@ -0,0 +1,76 @@
+//! Implements the function environment (e.g. a name-to-function mapping) for interpretation.
+
+use cranelift_codegen::ir::{FuncRef, Function};
+use std::collections::HashMap;
+
+#[derive(Default)]
+pub struct Environment {
+    functions: HashMap<FuncRef, Function>,
+    function_name_to_index: Vec<String>,
```

Comment:
It looks like this is actually mapping indices to names. If so, then the field name should be changed to `index_to_function_name` to reflect that.